### PR TITLE
Add a draft article for FTW-product related changes

### PR DESCRIPTION
**Pull request by Gnito**
1. This adds a new article ftw-introduction/**ftw-product**
2. I scanned through Tutorial articles to check if there are needs for changes
    - 2 advanced articles (_Customize pricing_ and _Create a new transaction process_) were too tied to FTW-daily but other articles were more or less understandable with couple of `<extrainfo>` comments.
3. And for update needs to other articles I searched for
    - "FTW-hourly"
    - "config.js"
    - "marketplace-custom-config.js"
    - "default-location-searches.js"
    - "routeConfiguration.js"
    - "Routes.js"

### CODE REVIEW

File: src/docs/ftw-introduction/ftw-product/index.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+slug: ftw-product
+updated: 2021-03-11
+category: ftw-introduction
+ingress:
+  This article introduces FTW-product and what's different compared to
+  FTW-daily and FTW-hourly.
+published: true
+---
+
+FTW-product is a new Flex Template for Web. It's build with product
```
Comment (kpuputti) [inline src/docs/ftw-introduction/ftw-product/index.md:12]:
```suggestion
FTW-product is a new Flex Template for Web. It's built with product
```

File: src/docs/ftw-introduction/ftw-product/index.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+marketplaces in mind.
+
+## Differences in directory structure
+
+With this template, we decided to take more out of the possibilities
```
Comment (kpuputti) [inline src/docs/ftw-introduction/ftw-product/index.md:17]:
```suggestion
With this template, we decided to take advantage of the possibilities
```

File: src/docs/ftw-introduction/ftw-product/index.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+that code-splitting made possible. FTW-daily and FTW-hourly were created
+before code-splitting was possible, and pushing most of the components
+to a specific components directory (_src/components/_) made sense. It
```
Comment (kpuputti) [inline src/docs/ftw-introduction/ftw-product/index.md:20]:
```suggestion
to a specific `src/components` directory made sense. It
```

File: src/docs/ftw-introduction/ftw-product/index.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+was a clear place where customizers could start looking for a components
```
Comment (kpuputti) [inline src/docs/ftw-introduction/ftw-product/index.md:21]:
```suggestion
was a clear place where customizers could start looking for components
```

File: src/docs/ftw-introduction/ftw-product/index.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+under page-directories. Here are some of the relevant changes in the
+directory structure:
+
+- **src/config/**: Generic configuration files are moved to this
+  directory
+  - Extra configs:
+    - **mainSearchType** in Topbar. It's either _'keywords'_ or
+      _'location'_.
+    - **searchPageVariant**: _'list'_ & _'map'_.
+    - **listingPageVariant**: _'product'_ & 'TODO: hero'
```
Comment (kpuputti) [inline src/docs/ftw-introduction/ftw-product/index.md:55]:
```suggestion
    - **listingPageLayout**: _'full-image'_ & _'hero-image'_
```

This is the config now:

```js
// ListingPage has 2 layout options: 'hero-image' and 'full-image'.
// - 'hero-image' means a layout where there's a hero section with cropped image in the beginning of the page
// - 'full-image' shows image carousel, where listing images are shown with the original aspect ratio
const listingPageLayout = 'full-image';
```

File: src/docs/ftw-introduction/ftw-product/index.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+  - Renamed configs:
+    - **transactionProcessAlias** (in other templates:
+      bookingProcessAlias)
+    - **lineItemUnitType** (in other templates: bookingUnitType)
+- **src/routing**: Route configuration and UI components related to
+  routing are moved to this directory
+- UI components that are only used in a single page or within a more
+  complex component are nested within that page or component. For
+  example:
+  - _EditListinWizard_ is moved under _EditListinPage_
```
Comment (kpuputti) [inline src/docs/ftw-introduction/ftw-product/index.md:65]:
```suggestion
  - _EditListingWizard_ is moved under _EditListingPage_
```

File: src/docs/ftw-introduction/ftw-product/index.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+  - Search map and filters are moved under _SearchPage_
+  - _TransactionPage_ got subcomponents like _ActivityFeed_,
+    _ReviewModal_, _ReviewForm_, _SendMessageForm_, and
+    _TransactionPanel_.
+- **src/forms**: this directory was removed as there was only one form
+  that was truly shared between pages: _StripeConnectAccountForm_. It's
+  moved to _src/components/_.
+
+TODO: add other notable changes here
```
Comment (kpuputti) [inline src/docs/ftw-introduction/ftw-product/index.md:74]: Do we need to mention that the booking breakdown is now handled by the more generic `OrderPanel` that renders the new `ProductOrderForm` or the old `BookingDatesForm`?

File: src/docs/ftw-introduction/ftw-product/index.md [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+Roughly, the transaction process graph looks like this:
+
+![flex-product-default-process](./flex-product-default-process.png)
+
+## Managing stock
+
+Stock management is pretty straightforward. On EditListingWizard, (which
+is a component to create and update listings), provider can set the
+stock quantity for the listing. Under the hood, FTW-product calls API
+with _`sdk.stockAdjustments/compareAndSet()`_ with the value provider is
```
Comment (kpuputti) [inline src/docs/ftw-introduction/ftw-product/index.md:94]: This should be updated to the latest SDK API.

File: src/docs/ftw-introduction/ftw-product/index.md [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+trying to set (`newTotal`) and also the previous value (`oldTotal`). At
+the time of update, if the assumed old total still matches with the
+current stock, the **currentStock** entity is updated to the value
+passed through with `newTotal` parameter. This check is done to prevent
+race conditions, where someone orders a listing while provider is
+updating its stock.
+
+Listing entity has a relationship to "currentStock" entity - so,
+currentStock info can be included listing queries. Then on ListingPage,
```
Comment (kpuputti) [inline src/docs/ftw-introduction/ftw-product/index.md:103]: Does it need to be explicitly included anymore?

File: src/docs/ftw-introduction/ftw-product/index.md [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+customer can order as many listings as there is stock left for it.
+
+> Note: Flex API and FTW-product don't yet support listing variants (the
+> stock of yellow t-shirts vs red t-shirts). So, listing variants need
+> to be modelled with separate listings.
+
+## Shipping the product
+
+For product marketplaces, it's usually necessary to ask shipping address
+from customer. FTW-product ask shipping method on ListingPage and if
```
Comment (kpuputti) [inline src/docs/ftw-introduction/ftw-product/index.md:113]:
```suggestion
from customer. FTW-product asks shipping method on ListingPage and if
```

File: src/docs/ftw-introduction/ftw-product/index.md [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+'shipping' is chosen, shipping address is asked on checkout page. The
```
Comment (kpuputti) [inline src/docs/ftw-introduction/ftw-product/index.md:114]: For consistency:

```suggestion
'shipping' is chosen, shipping address is asked on CheckoutPage. The
```

File: src/docs/ftw-introduction/ftw-product/index.md [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+_request-payment_ and _request-payment-after-enquiry_.
+
+On TransactionPage (the page opened by clicking inbox items), shipping
+address or pickup address is shown to both parties of the transaction.
+Provider has 2 weeks to mark the item shipped
+(_transition/mark-delivered_) or the automatic expiration happens
+(_transition/auto-cancel_) and customer is refunded. If needed, customer
+can dispute the order from "Delivered" state.
+
+![Dispute modal](./dispute-modal.png)
```
Comment (kpuputti) [inline src/docs/ftw-introduction/ftw-product/index.md:125]: This was the first mention of Sneakertime. Should be briefly introduce the concept in the beginning of the article? Something like:

> While FTW daily was about renting saunas and FTW hourly was about booking Yoga teachers, FTW product is a conceptual marketplace called Sneakertime to buy sneakers.

File: src/docs/ftw-introduction/ftw-product/index.md (hunk 9 above)
Comment (Gnito) [reply src/docs/ftw-introduction/ftw-product/index.md:103]: Yes. E.g. ListingPage.duck.js
`include: ['author', 'author.profileImage', 'images', 'currentStock'],`

**Pull request merged**